.appveyor.yml

```diff
@@ -7,14 +7,19 @@
 
 environment:
   matrix:
-  - TOXENV: py35
-  - TOXENV: py36
-  - TOXENV: py37
+    - PYTHON: "C:\\Python35"
+    - PYTHON: "C:\\Python36"
+    - PYTHON: "C:\\Python37"
+    - PYTHON: "C:\\Python38"
 
 build: off
 
 install:
-  - pip install tox
+  - set PATH=%PYTHON%;%PYTHON%\\Scripts;%PATH%
+  - python -m pip install -U pip
+  - python -m easy_install -U setuptools
+  - python -m pip install -U pyyaml
+  - python -m pip install -U lxml
```

Comment:
Summarize this into a single `pip` call? And is it really necessary to install pip and setuptools? After all, you are already using pip to install pip... :-)

Reply:
Something that came to my mind only now (sorry!): Can't we just let pip install all dependencies of BenchExec? On Travis and GitLab CI we do this with `pip install --user .`. Would of course be nice if we would not need to maintain the dependencies in several places. At least currently we don't have dependencies that would look problematic on Windows.

Btw.: `lxml` is only necessary for tests, but not for table-generator tests, so it could even be excluded. But if we adopt my latest suggestion, then it won't hurt if `lxml` is installed nevertheless.

Reply:
I added them in the beginning because they seemed to solve some error messages that were occuring (installing pip was used to update it to the newest version). Those error messages seem to be taken care of now any way though, so I could remove them and summarize the other two calls into a single one as you suggested.

When using the suggested command with the `--user` flag however, the same problem occurs that already occured when the install command for `lxml` was missing, so [the build fails](https://ci.appveyor.com/project/PhilippWendler/benchexec/builds/35339936). Strangely enough it only fails for Python 3.5 and 3.8, which was also the case [when not installing `lxml`](https://ci.appveyor.com/project/PhilippWendler/benchexec/builds/35291790). I actually can't really think of how this is happening, do you have any ideas? As already stated above, explicitly giving the calls for installing `lxml` fixes the problem.

Reply:
Note that in both cases, the actual testing phase with `setup.py nosetests` fails, because whenever `setup.py` is executed it ensures that its `setup_requires` dependencies are installed, and ours list `lxml`. `setup.py` uses `easy_install` to install any missing dependencies, which is an old and somewhat deprecation installation utility. `easy_install` downloads the source package of `lxml` and this fails because it bundles C code, which would need to be compiled into a binary library with a compiler, which is missing.

For Python 3.6 and 3.7 it seems that `lxml` is already installed in the Python environment by Travis and so not installation attempt is made.

When you install `lxml` explicitly with `pip`, you can see that `pip` downloads a binary package of `lxml` (`lxml-4.5.2-cp38-cp38-win32.whl`) instead and installation of this works without the need for a compiler.

I think we should probably modernize our `setup.py` somewhat, though I do not have a good overview over modern best practices for Python packages.

An easy solution is to to just explicitly install `lxml` with `pip` before doing `pip install .` (we could even add `--ignore-installed` to avoid any redundant installation if the package already exists).

For the same reason (preferring binary packages provided by Ubuntu instead of packages installed by `easy_install` or `pip`) our GitLab tests run in Docker containers with `lxml` preinstalled.

Reply:
I see, that makes sense. 
So I guess we should handle the updating of the `setup.py` in a separate issue? Those changes would probably exceed the initial intention of the PR to run the tablegenerator tests on Windows platforms.

So for now, I've added the installation of `lxml` before the call of `pip install .`. However, the [`--ignore-installed` flag](https://pip.pypa.io/en/stable/reference/pip_install/#options) seems to overwrite the installed package instead of simply not installing it if it's already installed. The standard behavior of `pip` is to leave already installed package alone if I understand correctly. For some reason, [`lxml` is still installed for Python 3.6](https://ci.appveyor.com/project/PhilippWendler/benchexec/builds/35358438/job/o0pw6ar8db51p6xu) and 3.7. Could it be possible that the directories of the preinstalled and newly installed versions are different? And should we do something about it? Installing the package, even if it's possible that it's already preinstalled somewhere, doesn't seem like a big deal to me - especially regarding our intention to modernize the `setup.py` anyway and remove the implicit installation afterwards again.

Any way, as soon as the review of the PR is finished, I'd start rebasing the branch to fix and squash the commit history a bit, in case nothing speaks against this.

Reply:
Thanks!

> So I guess we should handle the updating of the `setup.py` in a separate issue?

That was just background info for you, I don't think we need to handle this anytime soon.

> However, the [`--ignore-installed` flag](https://pip.pypa.io/en/stable/reference/pip_install/#options) seems to overwrite the installed package instead of simply not installing it if it's already installed.

Sorry, I read the docs wrong :-)

> And should we do something about it? Installing the package, even if it's possible that it's already preinstalled somewhere, doesn't seem like a big deal to me

Exactly. Would have been just a minor optimization, only worth it if it is easy.

So for me this is ready.